### PR TITLE
Moving pkg/kubectl/util/storage to staging

### PR DESCRIPTION
**Pull request by pswica**
-  Moves pkg/kubectl/util/storage to staging.
- Publishes this code to the new kubectl repository: https://github.com/kubernetes/kubectl/pkg/util/storage
- Code will be available for re-use after publishing.

/kind cleanup
/sig cli
/area kubectl
```release-note
NONE
```

### CODE REVIEW

File: staging/publishing/rules.yaml
Comment (nikhita) [inline staging/publishing/rules.yaml]: it looks like client-go is not a dependency yet, can we remove it?

File: staging/publishing/rules.yaml
Comment (nikhita) [inline staging/publishing/rules.yaml]: nit: to be alphabetically consistent, would be better to put `api` over `apimachinery` :stuck_out_tongue:

**Comment by nikhita** [comment]
Please also kubectl to https://github.com/kubernetes/kubernetes/blob/master/staging/publishing/import-restrictions.yaml

File: staging/publishing/rules.yaml
Comment (pswica) [reply staging/publishing/rules.yaml]: Great catch! Thanks for reviewing, I will change this.

File: staging/publishing/rules.yaml
Comment (pswica) [reply staging/publishing/rules.yaml]: I will fix this too (I was just copying rules from another PR--there are lots for this big move). Thanks again for your review

**Comment by pswica** [comment]
@nikhita can you help me understand this portion? I'm still new to this codebase and I don't want to update this .yaml in the wrong places.

> Please also kubectl to https://github.com/kubernetes/kubernetes/blob/master/staging/publishing/import-restrictions.yaml

**Comment by nikhita** [comment]
> @nikhita can you help me understand this portion? I'm still new to this codebase and I don't want to update this .yaml in the wrong places.

Sure! We want to make sure that certain (staging) packages don't end up importing other staging packges. For example, apimachinery should never depend on another staging repo. These import restrictions are enforced by that yaml file.

For instance, in this:

https://github.com/kubernetes/kubernetes/blob/4e0b76469ffbc4645c6a5d82dcca3390a37ea331/staging/publishing/import-restrictions.yaml#L33-L37

It shows that apimachinery should only depend on kube-openapi and klog (they have the `k8s.io` imports because they are in the `kubernetes` GitHub org, but they are not staging repos). If someone adds an import to, say client-go, the tests will fail.

For kubectl, we'll need to add something like https://github.com/kubernetes/kubernetes/pull/79207/files#diff-2c52da2e345c051a2fed16d27e7702a7 to make sure we don't accidentally add any unwanted imports. :)

**Comment by pswica** [comment]
@nikhita thanks so so much for your thoughtful reviews, I tried to make the suggested changes. You are the best.

**Comment by nikhita** [comment]
No, no, thank _you_ for making these changes! :heart:

File: staging/publishing/rules.yaml
Comment (pswica) [reply staging/publishing/rules.yaml]: I re-added this. I think with all of our code moves and re-merging this is now necessary, otherwise tests start failing.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/assign @nikhita 
for staging approval

**Comment by pswica** [comment]
thank you @soltysh!
I will hold my changes until other PRs with more files changed merge first.
/hold

File: staging/publishing/rules.yaml
Comment (nikhita) [inline staging/publishing/rules.yaml]: nit: extra newline :)

File: staging/publishing/rules.yaml
Comment (nikhita) [inline staging/publishing/rules.yaml]: This should be `release-1.15` instead of `master`

**Review by nikhita — COMMENTED** [review]
Two comments, lgtm otherwise.

File: staging/publishing/rules.yaml
Comment (nikhita) [reply staging/publishing/rules.yaml]: (related: created https://github.com/kubernetes/kubernetes/pull/79288 to fix rules for apimachinery and client-go)

Comment (pswica) [reply staging/publishing/rules.yaml]: Thanks @nikhita for all of your suggestions and fixes! I will definitely implement everything when it's time to merge this in.

Comment (pswica) [reply staging/publishing/rules.yaml]: @nikhita by the time it was time for me to merge these changes were already in place. Thanks again for reviewing

**Review by seans3 — CHANGES_REQUESTED** [review]
Please remove "staging/publishing/rules.yaml" from this PR. We shouldn't have to change this file anymore. Your changes to this file are only indentation changes, making the indentation inconsistent.

**Comment by pswica** [comment]
Okay I changed that file. I think that was a remnant of me editing that file before all the merging.

**Pull request merged**
